### PR TITLE
Name forms based on its form key

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | yes
| Fixed tickets | fixes --
| Related issues/PRs | --
| License | MIT
| Documentation PR | --

#### What's in this PR?

Name forms based on its form key.

#### Why?

Default way of defining forms.

### CODE REVIEW

**Comment by danrot** [comment]
Tested and works for me. @sulu/core-team Is that change ok for everybody?

File: UPGRADE.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ it is using.
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/form-1.0.xsd"
 >
-    <key>categories</key>
+    <key>category_details</key>
```
Comment (alexander-schranz) [inline UPGRADE.md]: Line 101 missing formKey should also be changed.

Also if we use details the route name should also be details not only detail: sulu_category.edit_form.detail **`s`**

```
return [
    (new Route('sulu_category.edit_form.details', '/details', 'sulu_admin.form'))
	        ->setOption('tabTitle', 'sulu_category.details')
	        ->setOption('resourceKey', 'categories')
	        ->setOption('formKey', 'category_details')
	        ->setOption('backRoute', 'sulu_category.datagrid')
];
```

File: src/Sulu/Bundle/MediaBundle/Resources/js/views/MediaDetails/MediaDetails.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {ResourceStore} from 'sulu-admin-bundle/stores';
 import {translate} from 'sulu-admin-bundle/utils';
 import MediaUploadStore from '../../stores/MediaUploadStore';
 import SingleMediaUpload from '../../containers/SingleMediaUpload';
-import mediaDetailStyles from './mediaDetail.scss';
+import mediasDetailStyles from './mediaDetails.scss';
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/js/views/MediaDetails/MediaDetails.js]: `mediaDetailsStyles`

File: src/Sulu/Bundle/PageBundle/Admin/PageAdmin.php [hunk 3]
```diff
@@ -168,7 +168,7 @@ public function getRoutes(): array
                 ->setResourceKey('pages')
                 ->setFormKey('page_settings')
                 ->setBackRoute(static::WEBSPACES_ROUTE)
-                ->setTabTitle('sulu_page.page_form_settings')
+                ->setTabTitle('sulu_page.settings')
```
[the remark this answers is not in the document]
Comment (danrot) [reply src/Sulu/Bundle/PageBundle/Admin/PageAdmin.php]: This translation seems still to be missing.

**Pull request merged**